src/Config_Command.php

```diff
@@ -349,6 +349,255 @@ private static function get_wp_config_vars() {
 		return array_merge( $wp_config_vars, $wp_config_constants, $wp_config_includes_array );
 	}
 
+	/**
+	 * Sets the value of a specific variable or constant defined in
+	 * wp-config.php file.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <key>
+	 * : Key for the wp-config.php variable or constant.
+	 *
+	 * <value>
+	 * : Value to set the wp-config.php variable or constant to.
+	 *
+	 * [--add]
+	 * : Add the value if it doesn't exist yet.
+	 * This is the default behavior, override with --no-add.
+	 *
+	 * [--raw]
+	 * : Place the value into the wp-config.php file as-is (executable), instead of as a quoted string.
+	 *
+	 * [--target=<target>]
+	 * : Target string to decide where to add new values.
+	 * Defaults to "/** Absolute path to the WordPress directory".
+	 *
+	 * [--placement=<placement>]
+	 * : Where to place the new values in relation to the target string.
+	 * ---
+	 * default: 'before'
+	 * options:
+	 *   - before
+	 *   - after
+	 * ---
+	 *
+	 * [--buffer=<buffer>]
+	 * : Buffer string to put between an added value and its target string.
+	 * The following escape sequences will be recognized and properly interpreted: '\n' => newline, '\t' => tab.
+	 * Defaults to two EOLs.
+	 *
+	 * [--type=<type>]
+	 * : Type of the config value to set. Defaults to 'all'.
+	 * ---
+	 * default: all
+	 * options:
+	 *   - constant
+	 *   - variable
+	 *   - all
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Set the WP_DEBUG constant to true.
+	 *     $ wp config set WP_DEBUG true --raw
+	 *
+	 * @when before_wp_load
+	 */
+	public function set( $args, $assoc_args ) {
+		$path = $this->get_config_path();
+
+		list( $key, $value ) = $args;
+		$type = Utils\get_flag_value( $assoc_args, 'type' );
+
+		$options = array();
```

Comment:
I think it'd be better and more transparent to set the `WPConfigTransformer` options explicitly and directly here, passing the defaults, eg `'add' => Utils\get_flag_value( $assoc_args, 'add', true ),`, which would also allow the names to be mapped.